src/astro/files/locations/s3.py

```diff
@@ -0,0 +1,49 @@
+import os
+from typing import Dict, List, Tuple
+from urllib.parse import urlparse, urlunparse
+
+from airflow.hooks.base import BaseHook
+
+from astro.constants import FileLocation
+from astro.files.locations.base import Location
+from astro.utils.dependencies import AwsBaseHook, BotoSession, s3
+
+
+class S3(Location):
+    """Handler S3 object store operations"""
+
+    location_type = FileLocation.S3
+
+    @staticmethod
+    def _parse_s3_env_var() -> Tuple[str, str]:
+        """Return S3 ID/KEY pair from environment vars"""
+        return os.environ["AWS_ACCESS_KEY_ID"], os.environ["AWS_SECRET_ACCESS_KEY"]
+
+    def get_transport_params(self) -> Dict:
+        """Structure s3fs credentials from Airflow connection.
+        s3fs enables pandas to write to s3
+        """
+        if self.conn_id:
+            # The following line raises a friendly exception
+            BaseHook.get_connection(self.conn_id)
+            aws_hook = AwsBaseHook(self.conn_id, client_type="S3")
+            session = aws_hook.get_session()  # type: ignore
+        else:
+            key, secret = S3._parse_s3_env_var()
```

Comment:
```suggestion
            key, secret = cls._parse_s3_env_var()
```

Reply:
That way if we rename the Class name anytime, we don't need to change it everywhere.

You can also use:

```suggestion
            key, secret = self._parse_s3_env_var()
```

Reply:
Thanks, nice! didn't know this.